Michael Ong iP

maikongeh opened the pull request:
pull request for week 3

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

maikongeh committed:
test

test

maikongeh committed:
level1

Greet, Echo, Exit

maikongeh committed:
level 2

add, list

maikongeh committed:
level 4

Mark as done, todo, event deadline inherits tasks.

maikongeh committed:
level 6

delete, error handling and running test cases

maikongeh committed:
Level 7

Add save function

tashawan23 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+        System.out.println("____________________________________");
+
+        Scanner input = new Scanner(System.in);
+        while(input.hasNextLine()) {
```
Maybe it would be better if you avoid deep nesting to improve code quality

pyuxiang replied:
Another space missed 😉 

```suggestion
        while (input.hasNextLine()) {
```

tashawan23 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+        for(String str: strings) {
+            char identifier = str.charAt(1);
+            switch(identifier) {
+                case 'D':
```
Maybe it would be better to leave out the indentation for case clauses

tashawan23 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+        catch (FileNotFoundException e){
+            throw new FileNotFoundException("No File Detected");
+        }
+        System.out.println("____________________________________");
```
Would it be better to have this line format as a constant since it is repeated several times in the class?

tashawan23 commented on src/main/java/Database.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public ArrayList<String> readFile() throws FileNotFoundException {
+        try {
```
I like your indentations in this class!

tashawan23 reviewed (commented):
Overall, I found your code quite easy to understand for the most part except for a few places where nesting was too deep. I have noted some repeated code and coding standard violations. :)

pyuxiang commented on src/main/java/Database.java:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+
+    public  void writeTaskToFile(List<Task> tasks){
```
Be careful of the spaces 😉

```suggestion
    public void writeTaskToFile(List<Task> tasks){
```

pyuxiang commented on src/main/java/Database.java:
```diff
@@ -0,0 +1,47 @@
+
+    String name;
+
+
```
You might have inadvertently added an extra blank line here:
https://se-education.org/guides/conventions/java/intermediate.html#layout
```suggestion
```

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+        Database database = new Database( "data.txt");
+        ArrayList<String> listOfTasks;
+        ArrayList<Task> myList;
+       try {
```
The indentation is a little off here: should be in line with the block.
```suggestion
        try {
```

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+       try {
+            listOfTasks = database.readFile();
+            myList = readInput(listOfTasks);
+        }
+        catch (FileNotFoundException e){
```
The ``catch`` keyword should be in line with the closing brace.

```suggestion
        } catch (FileNotFoundException e){
```

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+        Scanner input = new Scanner(System.in);
+        while(input.hasNextLine()) {
+            String s = input.next();
+            switch (s){
+                case "todo":
+                    System.out.println("____________________________________");
```
The ``case`` statement should be in line with the ``switch``, and the case blocks one indentation lower.
```suggestion
            switch (s){
            case "todo":
                System.out.println("____________________________________");
```

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+                            throw new DukeException("Enter a valid todo");
+                        } else {
+                            char[] chars = s1.toCharArray();
+                            if(chars[1] == ' '){
```
Another missing space 😉

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+                            if(chars[1] == ' '){
+                                throw new DukeException("Enter valid todo");
+                            } else {
+                                try{
```
And another 😉

pyuxiang replied:
Applies to a couple other lines below.

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+                case "todo":
+                    System.out.println("____________________________________");
+                    try {
+                        String s1 = input.nextLine();
```
You might have missed this: variable names could do with a more specific name, especially since the scope isn't small. Perhaps `s1` can be renamed as `userInputLine`.

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+                                    database.writeTaskToFile(myList);
+
+                                }
+                            } catch(DukeException e) {
```
Not a coding standard, but figured I could chip in. Since the `catch` block is the same as that below in line 86, you can consider removing the `try-except` block here.

pyuxiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,261 @@
+                case "list":
+                    System.out.println("____________________________________");
+                    System.out.println("Here are the tasks in your list:");
+                    for(int i=1; i< myList.size()+1; i++){
```
Note the spaces here as well:
```suggestion
                    for (int i = 1; i < myList.size() + 1; i++) {
```

pyuxiang commented on src/main/java/Database.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class Database {
```
Consider adding a couple of class-level comments describing the class.
Might not be immediately clear otherwise what this `Database` class is intended for.

pyuxiang commented on src/main/java/Database.java:
```diff
@@ -0,0 +1,47 @@
+import java.io.File;
```
Your `import` statements are very nicely formatted 🎉

pyuxiang reviewed (commented):
A couple of code standard comments, but otherwise adheres to standards pretty nicely!

maikongeh committed:
Level-8

Implement date function using LocalDate package

maikongeh committed:
A-MoreOOP

implement database - same functionality as storage,
parser to handle different switch cases, Ui to handle ouput for different commands and error handling.

maikongeh committed:
A-Packages

package files in duke

maikongeh committed:
Commit before taskList

s

maikongeh committed:
Implement TaskList

maikongeh committed:
A-JavaDoc

Introduce JavaDocs to my class and methods

maikongeh committed:
A-CodingStandard

Adhere to coding standards

maikongeh committed:
Merge branch 'Branch-A-CodingStandard'

maikongeh committed:
Level-9

Implemented find

maikongeh committed:
Merge branch 'branch-Level-9'

maikongeh committed:
Merge branch 'add-gradle-support'

maikongeh committed:
A-Gradle

Implemented gradle support

maikongeh committed:
Merge branch 'branch-A-Gradle'

maikongeh committed:
Save Before GUI

maikongeh committed:
Level-10

implement GUI

maikongeh committed:
Merge branch 'branch-Level-10'

maikongeh committed:
A-Assertions

implement Assertions

maikongeh committed:
Merge branch 'branch-Assertions'

maikongeh committed:
A-UserGuide

Create User guide in ReadMe

maikongeh committed:
fix bugs found

maikongeh committed:
update final file

maikongeh committed:
update

maikongeh committed:
improve codeQuality

maikongeh committed:
Merge branch 'branch-A-codeQuality'

maikongeh committed:
move Ui.png to ip folder

maikongeh committed:
Create config.yml file Set theme jekyll-theme-cayman

maikongeh committed:
Merge branch 'branch-UserGuide'

maikongeh committed:
Set theme jekyll-theme-cayman

maikongeh committed:
Set theme jekyll-theme-cayman

maikongeh committed:
Set theme jekyll-theme-cayman

maikongeh committed:
Update README.md

maikongeh committed:
Merge branch 'master' of https://github.com/maikongeh/ip

maikongeh committed:
reduce long methods and deep nesting

maikongeh committed:
Merge branch 'master' of https://github.com/maikongeh/ip
